scripts/categorizer.py

```diff
@@ -408,38 +414,54 @@ def main():
         is_assignment = determine_if_assignment(line_number, multiline_number, line, categorizations, is_function_def)
         is_function_call = determine_if_function_call(line_number, multiline_number, line, categorizations)
 
-        categorizations.append(LineAndCats(line=line, multiline_statement_number=multilines[line_number], comment=0,
-                                           conditional=is_conditional, empty=0, func_def=is_function_def, assignment=is_assignment, func_call= is_function_call))
+        categorizations.append(LineAndCats(line=line, multiline_statement_number=multilines[line_number],
+                                           conditional=is_conditional, func_def=is_function_def,
+                                           assignment=is_assignment, func_call= is_function_call))
 
-        categorizations_numbers = range(0, 8)
-        hash_string = "%s" % line
 
         hash_object = hashlib.md5((line + " " + str(line_number)).encode())
 
-        # String plus categorization
-        for number in categorizations_numbers:
-            hash_string += " " + str(categorizations[-1][number])
+        # Create a string which tracks line classfications/categorizations to store in hash
+        line_classifications = ""
+        for number in range(len(LineAndCats._fields)):
+            line_classifications += " " +  "%s " % (LineAndCats._fields[number]) + str(categorizations[-1][number])
 
-        hash_storage[hash_object.hexdigest()] = hash_string
+        hash_storage[hash_object.hexdigest()] = line_classifications
 
-    # Debug portion
+    # Debug portion -> maybe have all of theese produced into folders when ran???
     print_all_cats(lines, "func_call", categorizations)
 
-    did_something_change = False
+    print("Uncategorized LInes")
+    # TODO change editor background to be more friendly with black
+    # Print out lines which are not categorized as anything yet..
+    for cat in categorizations:
+        for number in range(1, len(LineAndCats._fields)):
+            if cat[number]:
```

Comment:
Is that category set to true once it's completed? I'm not sure I understand the logic here, maybe add a brief comment?

Reply:
I agree this section and the following section are obtuse. The default for each field is zero. if it is not zero then this for loop does not flag it as uncategorized.

Reply:
Excuse me, it does not flag it. It prints out uncategorized lines.